vfs/directory.go

```diff
@@ -257,6 +335,45 @@ func bulkUpdateDocsPath(c *Context, olddoc *DirDoc, newpath string) error {
 	return err
 }
 
+func fetchChildren(c *Context, parent *DirDoc) (files []*FileDoc, dirs []*DirDoc, err error) {
+	var docs []json.RawMessage
+	sel := mango.Equal("folder_id", parent.ID())
+	req := &couchdb.FindRequest{Selector: sel, Limit: 10}
+	err = couchdb.FindDocs(c.db, FsDocType, req, &docs)
+	if err != nil {
+		return
+	}
+
+	var t struct {
+		Type string `json:"type"`
+	}
+
+	// @NOTE: this unmarshaling technique is very slow...
```

Comment:
How about having a common struct for unmarshalling, that can then be converted to File or Directory ?